packages/data/lib/database/dao/movie_dao.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:data/database/entities/movie.dart';
+import 'package:floor/floor.dart';
+
+@dao
+abstract class MovieDao {
+  @Query('SELECT * FROM Movie WHERE moviesType = :movieType')
+  Future<List<Movie>> findMoviesByType(int movieType);
+
+  @Query('DELETE FROM Movie WHERE moviesType = :movieType')
+  Future<void> deleteAllMoviesByType(int movieType);
```

Comment:
when u receive the new list, don't delete all movies of the type from db. delete only the movies which are not actual

Reply:
@sergei-mikhailovskii-idf  I think I cannot control order of movies from api, so I remove all movies of particular type from db and insert new in order which is defined by api.

Reply:
Fixed.